libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -703,21 +703,9 @@ method onTopicSubscription*(g: GossipSub, topic: string, subscribed: bool) =
     # Send unsubscribe (in reverse order to sub/graft)
     procCall PubSub(g).onTopicSubscription(topic, subscribed)
 
-method publish*(
-    g: GossipSub, topic: string, data: seq[byte]
-): Future[int] {.async: (raises: []).} =
-  logScope:
-    topic
-
-  if topic.len <= 0: # data could be 0/empty
-    debug "Empty topic, skipping publish"
-    return 0
-
-  # base returns always 0
-  discard await procCall PubSub(g).publish(topic, data)
-
-  trace "Publishing message on topic", data = data.shortLog
-
+proc collectPeersForPublish(
+    g: GossipSub, topic: string, payloadLen: int
+): Result[HashSet[PubSubPeer], void] {.raises: [].} =
```

Comment:
there's no need for Result here - using Result just introduces additional copies without providing any new semantic information (the calling code can equally easy check the length) - also, if we were to use it, we typically spell `Result[T, void]` as `Opt[T]`.

Also, finding no peers in the mesh is not an error - it's a normal condition from which the protocol is designed to recover